metadata-ingestion/src/datahub/ingestion/source/identity/azure_ad.py

```diff
@@ -270,16 +270,33 @@ def _get_azure_ad_data(self, kind: str) -> Iterable[List]:
                 self.report.report_failure("_get_azure_ad_data_", error_str)
                 continue
 
+    def _map_identity_to_urn(self, func, id_to_extract, mapping_identifier):
+        result, error_str = None, None
+        try:
+            result = func(id_to_extract)
+        except ValueError as e:
+            error_str = (
+                "Failed to extract DataHub {} from Azure AD User {} due to '{}'".format(
+                    mapping_identifier, id_to_extract.get("displayName"), repr(e)
+                )
+            )
+        if not result:
+            error_str = "Failed to extract DataHub {} from Azure AD User {} unkown reason".format(
+                mapping_identifier, id_to_extract.get("displayName")
+            )
+        if error_str is not None:
+            logger.error(error_str)
+            self.report.report_failure(mapping_identifier, error_str)
+        return result, error_str
+
     def _map_azure_ad_groups(self, azure_ad_groups):
         for azure_ad_group in azure_ad_groups:
-            corp_group_urn = self._map_azure_ad_group_to_urn(azure_ad_group)
-            if not corp_group_urn:
-                error_str = (
-                    "Failed to extract DataHub Group Name from Azure Group for group named {}. "
-                    "Skipping...".format(azure_ad_group.get("displayName"))
-                )
-                logger.error(error_str)
-                self.report.report_failure("azure_ad_group_mapping", error_str)
+            corp_group_urn, error_str = self._map_identity_to_urn(
+                self._map_azure_ad_group_to_urn,
+                azure_ad_group,
+                "azure_ad_group_mapping",
+            )
+            if error_str is not None:
```

Comment:
Should we log any warning here?

Reply:
The next statement is `logger.error(error_str)` so we are logging error

Reply:
No - that line got removed in the diff right?